api/client/proxy/proxy_test.go

```diff
@@ -90,7 +112,28 @@ func TestGetProxyAddress(t *testing.T) {
 				t.Setenv(env.name, env.val)
 			}
 			p := GetProxyAddress(tt.targetAddr)
-			require.Equal(t, tt.proxyAddr, p)
+			if tt.proxyAddr == "" {
+				require.Nil(t, p)
+			} else {
+				require.Equal(t, tt.proxyAddr, p.Host)
+			}
 		})
 	}
 }
+
+func TestProxyAwareRoundTripper(t *testing.T) {
+	t.Setenv("HTTP_PROXY", "http://localhost:8888")
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{},
+		Proxy: func(req *http.Request) (*url.URL, error) {
+			return httpproxy.FromEnvironment().ProxyFunc()(req.URL)
+		},
+	}
+	rt := NewHTTPFallbackRoundTripper(transport, true)
+	req, err := http.NewRequest(http.MethodGet, "https://localhost:9999", nil)
+	require.NoError(t, err)
+	// Don't care about response, only if the scheme changed.
+	_, err = rt.RoundTrip(req)
```

Comment:
Assuming this will work.
```suggestion
	_, err = rt.RoundTrip(req) //nolint:bodyclose
```